### PR TITLE
making dependency_merge.yml same as other z-stream branches

### PR DESCRIPTION
**Pull request by omkarkhatavkar**
### Problem Statement
as dependency_merge.yml using same configuration as the master causing failure in the automerge.   

### Solution
making dependency_merge.yml same as other z-stream branches

### Related Issues
https://github.com/SatelliteQE/robottelo/actions/runs/7720921376

### CODE REVIEW

File: .github/workflows/dependency_merge.yml [hunk 1]
```diff
@@ -51,10 +37,10 @@ jobs:
 
       - name: Wait for other status checks to Pass
         id: waitforstatuschecks
-        uses: lewagon/wait-on-check-action@v1.3.3
+        uses: lewagon/wait-on-check-action@v1.3.1
```
Comment (Gauravtalreja1) [inline .github/workflows/dependency_merge.yml:40]: Can we keep it latest?
```suggestion
        uses: lewagon/wait-on-check-action@v1.3.3
```

File: .github/workflows/dependency_merge.yml [hunk 2]
```diff
@@ -51,10 +37,10 @@ jobs:
...
         with:
           ref: ${{ github.head_ref }}
-          repo-token: ${{ secrets.CHERRYPICK_PAT }}
+          repo-token: ${{ secrets.GITHUB_TOKEN }}
```
Comment (Gauravtalreja1) [inline .github/workflows/dependency_merge.yml:43]: is this change `s/CHERRYPICK_PAT/GITHUB_TOKEN` intended?

File: .github/workflows/dependency_merge.yml [hunk 3]
```diff
@@ -1,46 +1,32 @@
...
+on:
+  pull_request_target:
+    branches-ignore:
+      - master
 
 jobs:
   dependabot:
     name: dependabot-auto-merge
     runs-on: ubuntu-latest
-    if: github.event.pull_request.user.login == 'dependabot[bot]'
```
Comment (Gauravtalreja1) [inline .github/workflows/dependency_merge.yml:11]: can't we just add OR condition to check user is either  dependabot or Satellite bot here instead?

File: .github/workflows/dependency_merge.yml (hunk 2 above)
Comment (omkarkhatavkar) [reply .github/workflows/dependency_merge.yml:43]: making as uniform as https://github.com/SatelliteQE/robottelo/blob/6.14.z/.github/workflows/dependency_merge.yml please look at it

**Pull request merged**